docs: restructuring & introduction of mkdocs

marc0olo opened the pull request:
the current state can be viewed here: https://marc0olo.github.io/protocol/

I made some restructuring and added mkdocs to finally be able to find what somebody needs more easily.

I also noticed parts that definitely need to be reviewed again as they aren't really accurate any longer. but this should be done separately IMO.

this PR focuses on providing a guided and clickable structure for the protocol.

when this is merged a github action should automatically deploy the docs which should then be accesible at https://aeternity.com/protocol/

dincho commented:
What's the intention behind moving everything in a subdir ? While I agree it might look better from code perspective, unfortunately this is major BC break and I bet a lot of back-links will be broken after that. Also there are 14 pull requests here that this will create conflicts for.

So I'd suggest keeping the old file location for now at least.

marc0olo commented on mkdocs-hook.py:
```diff
@@ -0,0 +1,4 @@
+import shutil
```
will check that

marc0olo replied:
ok I remember now. at first I wanted to include it in the `.github/workflows` folder but somehow the path with the dot made problems there and then I decided to place it under root.

what about a `scripts` folder?

marc0olo replied:
I now have a specific mkdocs folder `00_mkdocs` where everything related to it is stored and being copied to via the hook

marc0olo commented:
> What's the intention behind moving everything in a subdir ? While I agree it might look better from code perspective, unfortunately this is major BC break and I bet a lot of back-links will be broken after that. Also there are 14 pull requests here that this will create conflicts for.
> 
> So I'd suggest keeping the old file location for now at least.

I generally agree. not 100% sure anymore why I did that. will try to handle this via the simple-hook.

independently I changed some structure and renamed index files in subfolders to `README.md`. should we also try to keep the filenames wherever possible? just wondering

marc0olo commented:
now using the old structure again. there are still some changes (e.g. `sync` folder and the renaming of the entrypoint markdowns to `README.md`)

can we keep that or should I revert that, too?

marc0olo commented:
improved the docs structure. using a separate `.docssite` folder now for the mkdocs stuff

dincho commented:
I'd suggest keeping the old file name as well and point it with link to the new index file (aka README.md)

dincho commented on AENS.md:
```diff
@@ -3,7 +3,7 @@
 ## Abstract
 
-This document describes the first draft of the Aeternity Naming System,
+This document describes the first draft of the AEternity Naming System,
```
wut ?

dincho replied:
I don't get it ? You want to emphasise on abbreviation? Even that does not makes much sense to me :)

marc0olo replied:
ahh I think this was updated. should be fixed already. but I will double check.

dincho commented on channels/OFF-CHAIN.md:
```diff
@@ -853,7 +853,7 @@ useable.
 Message code: 18
 
 Since both sides may initiate a `withdraw_created` request, it is possible
-that both may do so at roughly the same time. In particular, in the Aeternity node
```
I think there was a discussion about capitalisation, I'm definitely not an English expert, but did you double checked it should be lowercase ?

marc0olo replied:
to be discussed. maybe this needs to be addressed in another PR. we should be consistent here

dincho replied:
any update on this ? consistency can be accompanied by `aeternity` -> `Aeternity` as well.

marc0olo replied:
seems like I missed that one. will check again. thanks

dincho commented on contracts/events.md:
````diff
@@ -12,7 +11,7 @@ The events are stored in the [contract call object](../serializations.md#contrac
 
 Let's consider the following Sophia example:
-```
+```sophia
````
I doubt github highlighter supports that, if you find a closer to syntax language that would be better isn't it ? If not, this is better than missing annotation I guess.

marc0olo replied:
well I hope I can push this topic forward. there is a quite old PR about this. so I wanted at least to prepare the code, see https://github.com/github/linguist/pull/4719.

we need to revisit this.

dincho commented on node/api/README.md:
```diff
@@ -235,15 +247,4 @@ The `'data'` element may contain more detailed information inside `'data'`:
-* [Contract API usage](./contract_api_usage.md)
-* [Channels API usage](./channels_api_usage.md)
-* [Mining API usage](./mining_api_usage.md)
+Detailed message transcripts from test suites can be found [here](https://github.com/aeternity/protocol/tree/master/specification/node/api/examples/channels/json-rpc).
```
why absolute URL ?

marc0olo replied:
will be fixed :D
